src/glass/src/app/pages/deployment-page/deployment-page.component.ts

```diff
@@ -103,32 +106,26 @@ export class DeploymentPageComponent implements OnInit {
         if (success) {
           this.pollAssimilationStatus();
         } else {
-          this.handleError(undefined, 'Failed to start device deployment.');
+          this.handleError('Failed to start device deployment.');
         }
       },
-      () => {
-        this.handleError();
+      (err) => {
+        this.handleError(err);
       }
     );
   }
 
   pollAssimilationStatus(): void {
-    of(true)
-      .pipe(
-        delay(5000),
-        mergeMap(() => this.orchService.assimilateStatus())
-      )
+    this.orchService
+      .assimilateStatus()
+      .pipe(this.pollService.poll((res) => !res, undefined, 'Failed to deploy disks.'))
```

Comment:
We are using `devices` allover the page, so we should use it here, too.
```suggestion
      .pipe(this.pollService.poll((res) => !res, undefined, 'Failed to deploy devices.'))
```